fonduer/matchers.py

```diff
@@ -228,11 +228,18 @@ def init(self):
         self.attrib = self.opts.get('attrib', WORDS)
         self.sep = self.opts.get('sep', " ")
 
+        # Extending the RegexMatch to handle search(instead of only match)
+        # and adding a toggle for full span match.
+        # Default values are set to False and True for search flag and full
+        # span matching flag respectively.
+        self.search = self.opts.get('search', False)
+        self.full_match = self.opts.get('full_match', True)
```

Comment:
`self.full_match` is to toggle appending `$` to regex

Eg:

```
phrase = 'Invoice#:2387621387'
r1 = re.compile(r'Invoice')
r2 = re.compile(r'(Invoice)$')

r1.search(phrase) # returns <_sre.SRE_Match object; span=(0, 7), match='Invoice'>
r2.search(phrase) # returns None
```
This is happening because $ matches the end of the string but the expression can be part of the span not at the end.